tests/components/zerproc/test_light.py

```diff
@@ -0,0 +1,316 @@
+    mock_state_2 = pyzerproc.LightState(True, (0, 80, 255))
+
+    with patch(
+        "homeassistant.components.zerproc.light.pyzerproc.discover",
+        return_value=[mock_light_1, mock_light_2],
+    ), patch.object(mock_light_1, "connect"), patch.object(
+        mock_light_2, "connect"
+    ), patch.object(
+        mock_light_1, "get_state", return_value=mock_state_1
+    ), patch.object(
+        mock_light_2, "get_state", return_value=mock_state_2
+    ):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+    state = hass.states.get("light.ledblue_ccddeeff")
+    assert state.state == STATE_OFF
+    assert state.attributes == {
+        ATTR_FRIENDLY_NAME: "LEDBlue-CCDDEEFF",
+        ATTR_SUPPORTED_FEATURES: SUPPORT_BRIGHTNESS | SUPPORT_COLOR,
+    }
+
+    state = hass.states.get("light.ledblue_33445566")
+    assert state.state == STATE_ON
+    assert state.attributes == {
+        ATTR_FRIENDLY_NAME: "LEDBlue-33445566",
+        ATTR_SUPPORTED_FEATURES: SUPPORT_BRIGHTNESS | SUPPORT_COLOR,
+        ATTR_BRIGHTNESS: 255,
+        ATTR_HS_COLOR: (221.176, 100.0),
+        ATTR_RGB_COLOR: (0, 80, 255),
+        ATTR_XY_COLOR: (0.138, 0.08),
+    }
+
+    with patch.object(hass.loop, "stop"), patch.object(
+        mock_light_1, "disconnect"
+    ) as mock_disconnect_1, patch.object(
+        mock_light_2, "disconnect"
+    ) as mock_disconnect_2:
+        await hass.async_stop()
+
+    assert mock_disconnect_1.called
+    assert mock_disconnect_2.called
+
+
+async def test_discovery_exception(hass):
+    """Test platform setup."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    mock_entry = MockConfigEntry(domain=DOMAIN)
+    mock_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.zerproc.light.pyzerproc.discover",
+        side_effect=pyzerproc.ZerprocException("TEST"),
+    ):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+    # The exception should be captured and no entities should be added
+    assert len(hass.data[DOMAIN]["addresses"]) == 0
+
+
+async def test_connect_exception(hass):
+    """Test platform setup."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    mock_entry = MockConfigEntry(domain=DOMAIN)
+    mock_entry.add_to_hass(hass)
+
+    mock_light = pyzerproc.Light("AA:BB:CC:DD:EE:FF", "LEDBlue-CCDDEEFF")
+
+    with patch(
+        "homeassistant.components.zerproc.light.pyzerproc.discover",
+        return_value=[mock_light],
+    ), patch.object(
+        mock_light, "connect", side_effect=pyzerproc.ZerprocException("TEST")
+    ):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+    # The exception should be captured and no entities should be added
+    assert len(hass.data[DOMAIN]["addresses"]) == 0
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)